tasks/collect-slack-notification-params/collect-slack-notification-params.yaml

```diff
@@ -0,0 +1,185 @@
+      default: data.json
+    - name: hacUrl
+      type: string
+      description: Root URL of HAC UI (hybrid access console)
+      default: "https://console.redhat.com/preview/application-pipeline/workspaces"
+  workspaces:
+    - name: data
+      description: The workspace where the snapshot spec json file resides
+  results:
+    - name: message
+      type: string
+      description: the formatted message to use in downstream tasks
+    - name: slack-notification-secret
+      type: string
+      description: Name of secret which contains keyname=webhookURLs
+    - name: slack-notification-secret-keyname
+      type: string
+      description: Name of key within secret which contains webhook URL
+  steps:
+    - name: create-message
+      image: quay.io/hacbs-release/release-utils:85ab98a7ec63c3d8d9ec3c4982ff0e581bcb3c83
+      script: |
+        #!/usr/bin/env sh
+        set -x
+
+        DATA_FILE="$(workspaces.data.path)/$(params.dataPath)"
+        if [ ! -f "${DATA_FILE}" ] ; then
+            echo "No valid data file was provided."
+            exit 1
+        fi
+        SNAPSHOT_FILE="$(workspaces.data.path)/$(params.snapshotPath)"
+        if [ ! -f "${SNAPSHOT_FILE}" ] ; then
+            echo "No valid snapshot file was provided."
+            exit 1
+        fi
+        RELEASE_FILE="$(workspaces.data.path)/$(params.releasePath)"
+        if [ ! -f "${RELEASE_FILE}" ] ; then
+            echo "No valid release file was provided."
+            exit 1
+        fi
+
+        if [ $(jq '.slack | has("slack-notification-secret")' $DATA_FILE) == false ] ; then
+            echo "No secret name provided via 'slack.slack-notification-secret' key in Data."
+            echo -n "" > $(results.message.path)
+            echo -n "" > $(results.slack-notification-secret.path)
+            echo -n "" > $(results.slack-notification-secret-keyname.path)
+            exit
+        fi
+        slack_notification_secret=$(jq -r '.slack."slack-notification-secret"' $DATA_FILE)
+        echo -n "${slack_notification_secret}" > $(results.slack-notification-secret.path)
+
+        if [ $(jq '.slack | has("slack-webhook-notification-secret-keyname")' $DATA_FILE) == false ] ; then
+            echo "No secret key name provided via 'slack.slack-webhook-notification-secret-keyname' key in Data."
+            echo -n "" > $(results.message.path)
+            echo -n "" > $(results.slack-notification-secret-keyname.path)
+            exit
+        fi
+        slack_notification_secret_keyname=$(jq -r '.slack."slack-webhook-notification-secret-keyname"' $DATA_FILE)
+        echo -n "${slack_notification_secret_keyname}" > $(results.slack-notification-secret-keyname.path)
+
+        origin_namespace=$(jq -jr '.metadata.namespace' $RELEASE_FILE)
+        target_namespace=$(jq -r '.status.target' $RELEASE_FILE)
+
+        origin_workspace=$(echo $origin_namespace | sed 's/-tenant//g')   
```

Comment:
I would probably suggest using `${origin_workspace%-tenant}`, see "Substring Removal" here: https://tldp.org/LDP/abs/html/string-manipulation.html
But if you prefer to use `sed`, then at least add `$` to limit this substitution to the end of the string: `sed 's/-tenant$//g'`

The same applies to the next line as well.

Reply:
Fixed using the $ with sed